text/0000-light-dom-scoped-styles.md

```diff
@@ -0,0 +1,232 @@
+
+The main reason for this is that two component templates can contain `*.css` files that `@import` the same shared CSS file. This shared CSS file would need to be separately scoped for the two components. Since these shared CSS files could be very large (e.g. a CSS framework), and since they may `@import` further CSS files, it would be impractical to scope all of these files at compile time for each component. It would lead to a lot of duplicated code being generated on the server and sent to the client.
+
+Many of these design decisions were made to emulate native shadow DOM, which is perfectly content to `@import` shared CSS files within separate shadow roots, and to scope them accordingly. However, for light DOM style scoping, we don't have the same constraint of needing to match native shadow DOM semantics.
+
+So for light DOM style scoping, we can have a simpler system: disallow `@import` entirely, and evaluate the scoped CSS at compile time. This has performance benefits, at the cost of a probably-niche feature for developers.
+
+#### `lwc:dom="manual"` and `MutationObserver`
+
+Like the issue with `@import`, this one arises from the need to match native shadow DOM semantics. Consider this example:
+
+```html
+<!-- example.html -->
+<template>
+  <div lwc:dom="manual"></div>
+</template>
+```
+
+```css
+/* example.css */
+span {
+    color: green;
+}
+```
+
+```js
+// example.js
+class Example extends LightningElement {
+  renderedCallback() {
+    this.template.querySelector('div').appendChild(document.createElement('span'))
+  }
+}
+```
+
+In this case, the `<span>` is dynamically inserted into the component's shadow DOM, but synthetic shadow DOM needs some way to monitor the DOM for changes so that it can apply the styling token. This is where `lwc:dom="manual"` comes in – it's a signal to add a `MutationObserver` to track changes.
+
+This is a lot of extra machinery to support style scoping. Users need to know about `lwc:dom="manual"`, and the framework needs to create and disconnect a `MutationObserver`.
+
+We would like to avoid this for light DOM style scoping, so we have a simpler system: dynamically-inserted elements are not scoped. Incidentally, this is how Vue and Svelte scoped styles work – there's no expectation that you can mutate the DOM with vanilla DOM APIs and still have the scoping token applied.
+
+### Targeting the root element
+
+With both global and scoped light DOM styles, it is possible to style the component's root element using e.g.:
+
+```css
+x-mycomponent {}
+```
+
+Note that this means that, in scoped light DOM styles, a light parent and light child can both style the child's root element. For instance:
+
+```css
+/* parent.scoped.css */
+x-child {}
+```
+
+```css
+/* child.scoped.css */
+x-child {}
+```
+
+This will result in two scoping tokens being applied to the `<x-child>` element – one from the parent, and another from the child. Precendence order is not guaranteed.

```

Comment:
```suggestion
This will result in two scoping tokens being applied to the `<x-child>` element – one from the parent, and another from the child. Order of precedence is not guaranteed.

```